switch hex-rgb to polished parseToRgb

jameslnewell opened the pull request:
## Description
<!-- Please describe what you have changed or added -->
Temporary fix for the issue discussed in https://github.com/WordPress/gutenberg/pull/17963#discussion_r349374328

Replaced the `hex-rgb` package+function for `polished`'s `parseToRgb` function since the `hex-rgb` package doesn't provide any transpiled modules.

[Polished is 10kb gziped but highly treeshakable.](https://bundlephobia.com/result?p=polished@3.4.2)

## How has this been tested?
<!-- Please describe in detail how you tested your changes. -->
<!-- Include details of your testing environment, tests ran to see how -->
<!-- your change affects other areas of the code, etc. -->
Ran unit tests, checked the storybooks.

## Screenshots <!-- if applicable -->
No visual changes.

## Types of changes
<!-- What types of changes does your code introduce?  -->
<!-- Bug fix (non-breaking change which fixes an issue) -->
<!-- New feature (non-breaking change which adds functionality) -->
<!-- Breaking change (fix or feature that would cause existing functionality to not work as expected) -->
Bug fix.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. <!-- Check code: `npm run lint`, Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/javascript/ -->
- [x] My code follows the accessibility standards. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/accessibility-coding-standards/ -->
- [x] My code has proper inline documentation. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/javascript/ -->
- [x] I've included developer documentation if appropriate. <!-- Handbook: https://developer.wordpress.org/block-editor/ -->
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR. <!-- React Native mobile Gutenberg guidelines: https://github.com/WordPress/gutenberg/blob/master/docs/contributors/native-mobile.md -->.

closes #18719

Commit:
switch hex-rgb to polished parseToRgb

aduth commented:
One downside here is that it's about a 1400% size increase from `hex-rgb` ([0.7kb](https://bundlephobia.com/result?p=hex-rgb@4.1.0) to [10.6kb](https://bundlephobia.com/result?p=polished@3.4.2)).

I can plan to look a little closer in my morning. It'd be good to have this resolved in time for next week's release, so if it comes down to it, we can live with the larger bundle until another solution presents itself.

jameslnewell commented:
Yeah treeshaking isn't supported its a big hit. I can see a few alternatives but they're only 50% better:

- `color-convert` - [5kb, no treeshaking](https://bundlephobia.com/result?p=color-convert@2.0.1), updated 3 months ago
- `tinycolor2` - [5kb, no treeshaking](https://bundlephobia.com/result?p=tinycolor2@1.4.1), updated 3 years ago

youknowriad commented on packages/components/src/utils/colors.js:
```diff
@@ -2,7 +2,7 @@
  */
 import { get } from 'lodash';
-import hexRgb from 'hex-rgb';
+import { parseToRgb } from 'polished';
```
I think we already have tinycolor2 in our bundles. We could use it instead right?

ItsJonQ replied:
@jameslnewell Ah, you folks are so fast! You jumped on this before I could :)
I was going to suggest using `tinycolor2` since it was already there. My apologies for not realizing that when I submitted my original PR.

Thanks for the heads up @youknowriad + @aduth !

aduth commented:
>Polished is 10kb gziped but highly treeshakable.

The package is tree-shakeable, yes, but the 10kb number was the specific tree-shooken `parseToRgb` export (the full package is 34kb gzipped).

As mentioned, since we're already using `tinycolor2`, I think that's the way to go. I'll push up a change here shortly to help get the ball rolling in time for Monday's RC.

aduth committed:
Use tinycolor2 for hex-to-rgb conversion

aduth reviewed (approved):
Assuming latest changes look good to you, I think we can merge this 👍

aduth committed:
Update snapshot tests for the auto-incrementing IDs... I guess
